### ISSUE TITLE
TLC -dumpTrace option may save a different trace than MC.out

### ISSUE THREAD

**Issue by tangruize** [tlaplus/tlaplus#753]
## Description
I used TLC `-dumpTrace` option to run DieHard example to get the JSON trace, but found that the trace in the JSON file is different from MC.out.
A related issue: https://github.com/tlaplus/CommunityModules/issues/47#issuecomment-912226044

## Expected Behavior
Traces in MC.out, SpecTE and dumpTrace json/tla should be the same. 

## Actual Behavior
-dumpTrace may save a different trace.

## Steps to Reproduce
```sh
git clone https://github.com/tangruize/tlc-cmd
cd tlc-cmd/examples/DieHard
sed -i '/options/adump trace: true' DieHard.ini  # Add -dumpTrace options
python3 ../../tlcwrapper.py -m DieHard.ini  # -m to use community modules
ls -dt model* | head -1 | xargs -I '{}' python3 ../../trace_reader.py '{}'/MC.out | jq '.' > MC.out.json
ls -dt model* | head -1 | xargs -I '{}' jq . '{}'/MC_trace.json > dumpTrace.json
vim -O2 MC.out.json dumpTrace.json
# if they are the same, try again
```
![Screenshot_20220820_213949](https://user-images.githubusercontent.com/19245736/185748765-bfd79cf2-906b-415a-83b8-ce1b53eec9a8.png)

## Possible Fix
Maybe race condition?

## Your Environment
 - TLC version: 1.8.0

**Comment by lemmy** [comment]
How many TLC workers?

**Comment by tangruize** [comment]
4 workers.

**Comment by lemmy** [comment]
I agree that it would be better for `stdout/stderr` and `-dumptrace` to show the same counter-example.  However, I don't expect this to be an issue for most real-world specifications where it is unlikely for two or more workers to find a violation simultaneously.  In other words, this is a low-priority issue for me and likely won't be addressed unless a PR is contributed.

**Comment by lemmy** [comment]
Related: https://github.com/tlaplus/tlaplus/pull/1288
